app/src/main/java/com/github/se/polyfit/data/remote/firebase/PostFirebaseRepository.kt

```diff
@@ -1,19 +1,39 @@
 package com.github.se.polyfit.data.remote.firebase
 
 import android.util.Log
+import com.firebase.geofire.GeoFire
+import com.firebase.geofire.GeoLocation
+import com.firebase.geofire.GeoQueryEventListener
 import com.github.se.polyfit.model.post.Post
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.firestore.CollectionReference
 import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FieldPath
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.StorageReference
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
 
 class PostFirebaseRepository(
     private val db: FirebaseFirestore = FirebaseFirestore.getInstance(),
-    private val pictureDb: FirebaseStorage = FirebaseStorage.getInstance()
+    private val pictureDb: FirebaseStorage = FirebaseStorage.getInstance(),
```

Comment:
To be consistent with usage of Hilt in the rest of the app, it might be a smart idea to use dependency injection for both of the Firebase. The provides function should be added to the `UserModule.kt` file.